In [1]:
pip install apyori

In [2]:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from apyori import apriori

In [3]:
transactions = np.array((["A","B","C","E"], ["B","E"], ["C","D","E"],["A","C","D"],["A","C","E"]))

In [4]:
rules = apriori(transactions, min_support = 0.5, min_confidence = 0, min_lift = 0, min_length = 1)

In [5]:
results = list(rules)

In [6]:
results

[RelationRecord(items=frozenset({'A'}), support=0.6, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'A'}), confidence=0.6, lift=1.0)]),
 RelationRecord(items=frozenset({'C'}), support=0.8, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'C'}), confidence=0.8, lift=1.0)]),
 RelationRecord(items=frozenset({'E'}), support=0.8, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'E'}), confidence=0.8, lift=1.0)]),
 RelationRecord(items=frozenset({'C', 'A'}), support=0.6, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'C', 'A'}), confidence=0.6, lift=1.0), OrderedStatistic(items_base=frozenset({'A'}), items_add=frozenset({'C'}), confidence=1.0, lift=1.25), OrderedStatistic(items_base=frozenset({'C'}), items_add=frozenset({'A'}), confidence=0.7499999999999999, lift=1.2499999999999998)]),
 RelationRecord(items=frozenset({'C', 'E'}), support=0.6, ordered_statist

In [7]:
def inspect(results):
    rh          = [tuple(result[2][0][0]) for result in results]
    lh          = [tuple(result[2][0][1]) for result in results]
    supports    = [result[1] for result in results]
    confidences = [result[2][0][2] for result in results]
    lifts       = [result[2][0][3] for result in results]
    return list(zip(rh, lh, supports, confidences, lifts))

In [8]:
resultDataFrame = pd.DataFrame(inspect(results),
                columns=['rhs','lhs','support','confidence','lift'])

In [9]:
resultDataFrame


rhs     lhs  support  confidence  lift
0  ()    (A,)      0.6         0.6   1.0
1  ()    (C,)      0.8         0.8   1.0
2  ()    (E,)      0.8         0.8   1.0
3  ()  (C, A)      0.6         0.6   1.0
4  ()  (C, E)      0.6         0.6   1.0

Conclusión

Los niveles de K de soporte mas alto al que podemos llegar con estos datos teniendo un soporte minimo de 0.5 es:

Cuando K=1

            (A)
            Support:0.6
            Confidence:0.6
            Lift:1.0
            (C)
            Support:0.8
            Confidence:0.8
            Lift:1.0
            (E)
            Support:0.8
            Confidence:0.8
            Lift:1.0
                  

Cuando K=2

            (C,A)
            Support: 0.6
            Confidence: 0.6
            Lift: 1.0
            ________________________________
            (A->C)
            Support: 0.6
            Confidence: 1.0
            Lift: 1.25
            (C->A)
            Support: 0.6
            Confidence: 0.749999999999999
            Lift: 1.2499999999999998
            
            
            
         
            (E,C)
            Support: 0.6
            Confidence: 0.6
            Lift: 1.0
            _________________________________
            (E->C)
            Support: 0.6
            Confidence: 0.7499999999999999
            Lift:0.9374999999999998
            (C->E)
            Support: 0.6
            Confidence: 0.7499999999999999
            Lift=0.9374999999999998
            



